src/history/hs.c

```diff
@@ -1765,27 +1754,36 @@ __verify_history_store_id(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, uint32
         if (btree_id != this_btree_id)
             break;
 
+        /*
+         * If the stop time pair on the tombstone in the history store is already globally visible
+         * we can skip it.
+         */
+        if (__wt_txn_tw_stop_visible_all(session, &hs_cbt->upd_value->tw)) {
+            WT_STAT_CONN_INCR(session, cursor_prev_hs_tombstone);
+            continue;
+        }
+
         /*
          * If we have already checked against this key, keep going to the next key. We only need to
          * check the key once.
          */
         WT_ERR(__wt_compare(session, NULL, &hs_key, prev_hs_key, &cmp));
         if (cmp == 0)
             continue;
-        WT_WITH_PAGE_INDEX(session, ret = __hs_row_search(cbt, &hs_key, false));
-        WT_ERR(ret);
 
-        found = cbt->compare == 0;
-        WT_ERR(__cursor_reset(cbt));
+        WT_WITH_PAGE_INDEX(session, ret = __wt_row_search(cbt, &hs_key, false, NULL, false, NULL));
```

Comment:
Why did you switch from __hs_row_search to calling __wt_row_search directly? We have reduced the number of direct calls to __wt_row_search in the history store as much as possible in an effort to move towards having a standardized interface for history cursor operations.

Reply:
Here we are searching the data store, not the history store.

Reply:
Yes, it's a bug fix. Some change accidentally changed it from searching data file to searching history file before. As this part of the code is disabled so it is not picked by our testing.